### PR TITLE
Added Markdown Support in reviews

### LINKED ISSUE #8: Add Markdown support in reviews
Add github-flavored markdown support in writing (while posting) and wherever that review is rendered (review card and read more dialog). 

It needn't be too extensive, we should focus on baseline formatting features like lists, **bold**, *italic*, ~strikethrough~, headings etc.

## Must-haves
- We might in future add custom syntax (for mentioning users etc) if that aligns with the goals of this project. So to be prudent we should choose a markdown library that **supports extending custom syntax**.

- The implementation **must** support escaping characters.

- A `Preview` tab, just like here on github.

## Follow-ups
We can do many things to improve the UX:
- Add shortcut keys for inserting a markdown link syntax at current cursor.
- When typing a `*` or \`, the corresponding closing character can be automatically inserted rightwards to the cursor in the text area. 
- Pressing `enter` when writing a list should automatically insert `-` in the new line. (just like it happens here on github).
- Any other convenient behaviour.

We'd be fine if the implementation works mostly like github, since folks are used to writing here.

### PR DESCRIPTION
**Pull request by criticic**
Fixes #8

Uses the popular react-markdown library, which can also be extended to support additional custom syntax.

also added a preview tab when submitting/updating a review

### CODE REVIEW

**Comment by Varun-Kolanu** [comment]
image

image

The dashes should render bullet points right?

And on enter, bullets are not being automatically generated

**Comment by Varun-Kolanu** [comment]
@kuv2707 assigning to you. Ek baar intricacies test karna deployment preview se.

**Comment by criticic** [comment]
seems like a tailwind bug: https://jumpstartrails.com/discussions/solved-anyone-else-unable-to-get-simple-li-to-look-like-bullets

**Comment by kuv2707** [comment]
The generated html looks fine, but the required styling is absent.

image

image

Code blocks, list bullets are absent, though they are present in the generated html:

image

@criticic Can you fix the styling issue?

File: package.json [hunk 1]
```diff
@@ -15,9 +15,11 @@
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "react-hot-toast": "^2.4.1",
+    "react-markdown": "^9.0.1",
     "react-router-dom": "^7.0.2",
     "react-simple-star-rating": "^5.1.7",
-    "react-star-ratings": "^2.3.0"
+    "react-star-ratings": "^2.3.0",
+    "remark-gfm": "^4.0.0"
```
Comment (kuv2707) [inline package.json:22]: What are these needed for?

Comment (criticic) [inline package.json]: star ratings was present previously, remark-gfm is for github flavored markdown

File: package.json [hunk 2]
```diff
@@ -15,9 +15,11 @@
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "react-hot-toast": "^2.4.1",
+    "react-markdown": "^9.0.1",
     "react-router-dom": "^7.0.2",
     "react-simple-star-rating": "^5.1.7",
-    "react-star-ratings": "^2.3.0"
+    "react-star-ratings": "^2.3.0",
```
Comment (kuv2707) [inline package.json:21]: Why is this needed here?

Comment (kuv2707) [reply package.json:21]: It should not be in this PR i suppose

Comment (Varun-Kolanu) [reply package.json:21]: Wo shyd comma add hua isliye aagya PR me

**Comment by Varun-Kolanu** [comment]
@kuv2707 That html part was idk some tailwind bug as mentioned by @criticic 
Maybe something like adding custom css for `<li>` should work?

File: package.json (hunk 2 above)
Comment (kuv2707) [reply package.json:21]: O dikha nahi

**Comment by kuv2707** [comment]
Haa, custom css seems to be the most expedient solution. 

@criticic  Refer to [this](https://stackoverflow.com/questions/66356329/how-to-add-styling-for-elements-in-react-markdown). One of the solutions conflates tailwind with normal css, and that seems to be the most convenient way out of this.
